bin/ebuild.sh

```diff
@@ -186,12 +186,9 @@ export SANDBOX_ON=0
 # Ensure that $PWD is sane whenever possible, to protect against
 # exploitation of insecure search path for python -c in ebuilds.
 # See bug #239560, bug #469338, and bug #595028.
-if [[ -d ${HOME} ]]; then
-	# Use portage's temporary HOME directory if available.
-	cd "${HOME}" || die
-else
-	cd "${PORTAGE_PYM_PATH}" || \
-		die "PORTAGE_PYM_PATH does not exist: '${PORTAGE_PYM_PATH}'"
+# EAPI 8 requires us to use an empty directory here.
+if [[ -d ${PORTAGE_BUILDDIR}/empty ]]; then
```

Comment:
Apparently we can't cd into it for some internal Portage phases (e.g. `clean`) but I'm not sure if we need to care about the starting directory for them anyway.

Reply:
We need to care for every phase, due to the persistent threat of issues like [bug 239560](https://bugs.gentoo.org/239560) and so on. So, if `${PORTAGE_BUILDDIR}/empty` doesn't exist then we should keep the existing behavior:
```
cd "${PORTAGE_PYM_PATH}" || die "PORTAGE_PYM_PATH does not exist: '${PORTAGE_PYM_PATH}'"
```